field/src/field.rs

```diff
@@ -219,17 +219,17 @@ pub trait TwoAdicField: Field {
 
 /// An iterator over the powers of a certain base element `b`: `b^0, b^1, b^2, ...`.
 #[derive(Clone)]
-pub struct Powers<F: Field> {
-    base: F,
-    current: F,
+pub struct Powers<F> {
+    pub base: F,
```

Comment:
Hm this is meant to be instantiated as `generator.powers()`, would that cover the use case?

Reply:
Oh I see the valida code deals with an abstract field, how about moving `powers()` to `AbstractField`?